### PR TITLE
Upgrade go to 1.14 for Telemeter

### PR DESCRIPTION
**Pull request by kakkoyun**

### CODE REVIEW

File: ci-operator/config/openshift-priv/telemeter/openshift-priv-telemeter-master.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ build_root:
     cluster: https://api.ci.openshift.org
     name: release
     namespace: openshift
-    tag: golang-1.13
+    tag: golang-1.14
```
Comment (petr-muller) [inline ci-operator/config/openshift-priv/telemeter/openshift-priv-telemeter-master.yaml:12]: You should not modify `openshift-priv/telemeter` files but the `openshift/telemeter` ones. The changes will be propagated to `openshift-priv` automatically.

**Comment by kakkoyun** [comment]
@petr-muller Is it good proceed with this? Do I need to change anything else for the CI?
For more context, I have a PR requires this update https://github.com/openshift/telemeter/pull/338.

**Pull request merged**